controllers/contacts-controller.js

```diff
@@ -0,0 +1,82 @@
+import contactsService from "../models/index.js"
+
+import {HttpError} from "../helpers/index.js"
+
+import { contactAddSchema, contactUpdateSchema } from "../schemas/contact-schema.js";
+
+const getAll = async (req, res, next) => {
+    try {
+        const result = await contactsService.listContacts();
+        res.json(result);
+        
+    } catch (error) {
+       next(error);
+    }
+  };
+
+const getById = async (req, res, next) => {
+    try {
+        const {contactId} = req.params;
+        const result = await contactsService.getContactById(contactId);
+        if(!result) {
+            throw HttpError(404, "Not found");
+        }
+        res.json(result);
+    } catch (error) {
+       next(error)
+    }
+};
+
+const add = async (req, res, next) => {
+    try {
+        const {error} = contactAddSchema.validate(req.body);
+        if(error) {
+            throw HttpError(400, error.message);
+        }
```

Comment:
```suggestion
        if(error) {
            return next(HttpError(400, error.message));
        }
```
загалом вважається поганою практикою кидати помилку в try щоб таким чином перейти в catch

Reply:
прошу підказати як правильно, таким чином обробляти помилку викладач нам показав на лекції

Reply:
так теж можна, просто це не найкраща практика, web storm навіть буде позначати такий запис зауваженням

Reply:
return next(HttpError(400, error.message));

Reply:
можете так писати